### PR TITLE
Update inference docs

### PR DESCRIPTION
**Pull request by cdcapano**
This updates the documentation page on how to use `pycbc_inference`. Several things are updated:

 * I've added an explanation of the sampling and waveform transforms features in the config file syntax section.
 * I've added an example of how to run on a 2D normal distribution, and explain how to run on the other test distributions.
 * I've removed the deprecated `--skip-burn-in` option in the BBH and GW150914 examples, replacing instead with `--burn-in-function max_posterior`. I've also switched these from using `--niterations` to `n-independent-samples`, and have provided an explanation of what this means.

Example page is [here](https://www.atlas.aei.uni-hannover.de/~cdcapano/LSC/scratch/testdocs/latest/html/inference.html).

### CODE REVIEW

**Comment by cmbiwer** [comment]
Thanks for doing this. We should probably include documentation for the ``[constraint]`` sections but that doesn't need to be in this PR.

File: docs/inference.rst [hunk 1]
```diff
@@ -416,13 +512,11 @@ Now run::
         --output-file ${OUTPUT_PATH} \
         --processing-scheme ${PROCESSING_SCHEME} \
         --sampler kombine \
-        --skip-burn-in \
+        --burn-in-function max_posterior \
         --update-interval ${N_UPDATE} \
         --likelihood-evaluator gaussian \
-        --nwalkers ${N_WALKERS} \
```
Comment (soumide1102) [inline docs/inference.rst]: You've deleted the `--nwalkers` option.

**Comment by cdcapano** [comment]
@soumide1102 Oops, nice catch. I've fixed it.

**Comment by soumide1102** [comment]
The descriptions are nice.

**Pull request merged**